### PR TITLE
Dev offloading v10

### PR DESCRIPTION
**Pull request by regit**
This is an attempt to implement offloading/shunting of flows in suricata. Currently it only features an NFQ implementation of offloading mechanism. I will work later on an EBPF one for AF_PACKET.

PR builds:
- PR regit: https://buildbot.openinfosecfoundation.org/builders/regit/builds/175
- PR regit-pcap: https://buildbot.openinfosecfoundation.org/builders/regit-pcap/builds/171

### CODE REVIEW

**Comment by inliniac** [comment]
I think the term 'offload' is a bit unfortunate. It's way to broad and non-descriptive. I would prefer 'shunt' although I think it's not considered a nice word. I guess 'bypass' could be something, or perhaps 'flow bypass'. Iirc that is what we really mean right? Ask the capture method to stop sending us packets for a flow.

**Comment by regit** [comment]
I agree on 'bypass'. That will be better.

**Comment by inliniac** [comment]
I think we should hide this pattern in a function:
<pre>
if (p->livedev && p->livedev->offload_callback) {
    p->livedev->offload_callback(p);
}
</pre>

File: src/util-device.h [hunk 1]
```diff
@@ -20,27 +20,31 @@
...
 #include "queue.h"
 #include "unix-manager.h"
+#include "runmodes.h"
 
 /** storage for live device names */
 typedef struct LiveDevice_ {
     char *dev;  /**< the device (e.g. "eth0") */
+    enum RunModes runmode; /**< the runmode (e.g. "RUNMODE_NFLOG") */
     int ignore_checksum;
+    void (*offload_callback)(Packet *p);
```
Comment (inliniac) [inline src/util-device.h]: Our naming for functions and function pointers means this should be OffloadCallback

**Comment by inliniac** [comment]
This PR adds a lot of very verbose unittests with lots of code duplication. Can you guys clean that up as well? See 2c7cd1c22a6618dbc5207e927f3f99636c7fa204 as an example.

**Comment by inliniac** [comment]
Commit message " offloading: add crypted part " doesn't make much sense to me. It speaks about 'crypted files' ?

**Pull request closed without merging**